--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1507,7 +1507,17 @@ if(NOT DUCKDB_EXPLICIT_PLATFORM)
   set_target_properties(duckdb_platform_binary PROPERTIES RUNTIME_OUTPUT_DIRECTORY
                                                 ${PROJECT_BINARY_DIR})
   add_custom_command(
           OUTPUT ${PROJECT_BINARY_DIR}/duckdb_platform_out
           DEPENDS duckdb_platform_binary
-          COMMAND duckdb_platform_binary > ${PROJECT_BINARY_DIR}/duckdb_platform_out || ( echo "Provide explicit DUCKDB_PLATFORM=your_target_arch to avoid build-type detection of the platform" && exit 1 )
+          COMMAND ${CMAKE_COMMAND} -E rename 
+            ${PROJECT_BINARY_DIR}/duckdb_platform_binary.js 
+            ${PROJECT_BINARY_DIR}/duckdb_platform_binary.cjs || echo "No .js file to rename (not Emscripten build)"
+          COMMAND ${CMAKE_COMMAND} -E env 
+            ${CMAKE_COMMAND} -E echo "Running platform detection..."
+          COMMAND $<IF:$<BOOL:${EMSCRIPTEN}>,
+            node ${PROJECT_BINARY_DIR}/duckdb_platform_binary.cjs,
+            duckdb_platform_binary
+          > > ${PROJECT_BINARY_DIR}/duckdb_platform_out || ( echo "Provide explicit DUCKDB_PLATFORM=your_target_arch to avoid build-type detection of the platform" && exit 1 )
           WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
           )
 else()
